crates/sui-sdk/examples/utils/example_utils.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use reqwest::Client;
+use serde_json::json;
+use shared_crypto::intent::Intent;
+use sui_config::{sui_config_dir, SUI_KEYSTORE_FILENAME};
+use sui_keys::keystore::{AccountKeystore, FileBasedKeystore, InMemKeystore, Keystore};
+
+use sui_types::quorum_driver_types::ExecuteTransactionRequestType;
+
+use sui_sdk::rpc_types::SuiTransactionBlockResponseOptions;
+use sui_sdk::SuiClient;
+use sui_types::base_types::SuiAddress;
+use sui_types::digests::TransactionDigest;
+use sui_types::transaction::Transaction;
+
+// const SUI_FAUCET: &str = "https://faucet.devnet.sui.io/gas"; // devnet faucet
+
+const SUI_FAUCET: &str = "https://faucet.testnet.sui.io/gas"; // testnet faucet
+
+// if you use the sui-test-validator and use the local network; if it does not work, try with port 5003.
+// const SUI_FAUCET: &str = "http://127.0.0.1:9123/gas";
+
+#[allow(dead_code)]
+pub(crate) async fn sui_address_for_examples() -> Result<SuiAddress, anyhow::Error> {
+    let active_address: SuiAddress = SuiAddress::random_for_testing_only();
+    request_tokens_from_faucet(active_address).await?;
+    Ok(active_address)
+}
+
+#[allow(dead_code)]
+/// This function will try to use an active address from the wallet,
+/// and create another dummy address (in-memory) to be used in the examples.
+/// The purpose of this function is to facilitate running the examples without
+/// having to call the faucet every single time.
+/// When there is no wallet, it will create a wallet in the sui default config folder,
+/// and then create one address and it will request one coin from the faucet,
+/// for the set active address.
+pub(crate) async fn local_wallet_addresses(
+    sui: &SuiClient,
+) -> Result<Vec<SuiAddress>, anyhow::Error> {
+    let mut wallet = crate::utils::wallet::retrieve_wallet().await?;
+    // get the active address from the wallet
+    let active_address = wallet.active_address()?;
+    // get the SUI coins for this addressi
+    let coins = sui
+        .coin_read_api()
+        .get_coins(
+            active_address,
+            Some("0x2::sui::SUI".to_string()),
+            None,
+            None,
+        )
+        .await?;
+    // check if the address has coins, if not add some
+    if coins.next_cursor.is_none() {
+        request_tokens_from_faucet(active_address).await?;
+    }
```

Comment:
Better to do a balance check (check that there's sufficient SUI balance to run the examples), because otherwise the examples will still fail if the coin you found has a very low balance.